### PR TITLE
🏃Refactor kubeadm control plane reconcile funcs

### LINKED ISSUE #2065: Kubeadm-based control plane: Refactor reconcile functions
Some code in `reconcile` and `reconcileDelete` may be pulled up into the `Reconcile` function. Please see:
- https://github.com/kubernetes-sigs/cluster-api/pull/2037#discussion_r365606562
- https://github.com/kubernetes-sigs/cluster-api/pull/2037#discussion_r365606986

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2065

### CODE REVIEW

File: controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go [hunk 1]
```diff
@@ -105,13 +105,13 @@ func (r *KubeadmControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager, optio
...
 
-func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Result, _ error) {
+func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Result, reterr error) {
 	logger := r.Log.WithValues("kubeadmControlPlane", req.Name, "namespace", req.Namespace)
 	ctx := context.Background()
 
 	// Fetch the KubeadmControlPlane instance.
-	kubeadmControlPlane := &controlplanev1.KubeadmControlPlane{}
-	if err := r.Client.Get(ctx, req.NamespacedName, kubeadmControlPlane); err != nil {
+	kcp := &controlplanev1.KubeadmControlPlane{}
```
Comment (vincepri) [inline controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go]: For consistency with the other variables

File: controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go [hunk 2]
```diff
@@ -124,46 +124,52 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
...
+			reterr = kerrors.NewAggregate([]error{reterr, err})
+		}
+
 		// Always attempt to Patch the KubeadmControlPlane object and status after each reconciliation.
-		if patchErr := patchHelper.Patch(ctx, kubeadmControlPlane); patchErr != nil {
-			logger.Error(patchErr, "Failed to patch KubeadmControlPlane")
-			res.Requeue = true
+		if err := patchHelper.Patch(ctx, kcp); err != nil {
+			logger.Error(err, "Failed to patch KubeadmControlPlane")
+			reterr = kerrors.NewAggregate([]error{reterr, err})
```
Comment (vincepri) [inline controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go]: You can aggregate errors like this, it'll be expanded when the caller calls `Error()`

File: controlplane/kubeadm/controllers/kubeadm_control_plane_controller_test.go [hunk 3]
```diff
@@ -421,13 +421,10 @@ func TestReconcileNoClusterOwnerRef(t *testing.T) {
 		Log:    log.Log,
 	}
 
-	result, err := r.reconcile(context.Background(), kcp, r.Log)
+	result, err := r.Reconcile(ctrl.Request{NamespacedName: types.NamespacedName{Name: kcp.Name, Namespace: kcp.Namespace}})
 	g.Expect(err).NotTo(gomega.HaveOccurred())
 	g.Expect(result).To(gomega.Equal(ctrl.Result{}))
 
-	// Always expect that the Finalizer is set on the passed in resource
-	g.Expect(kcp.Finalizers).To(gomega.ContainElement(controlplanev1.KubeadmControlPlaneFinalizer))
```
Comment (vincepri) [inline controlplane/kubeadm/controllers/kubeadm_control_plane_controller_test.go]: If we can't find the cluster, the Reconcile function now returns early, and the finalizer isn't set. This is in line with what the other controllers are doing today

**Comment by dlipovetsky** [comment]
Thanks for cleaning this up @vincepri :smile:

**Pull request merged**